packages/graphqlgen/src/tests/typescript/__snapshots__/basic.test.ts.snap

```diff
@@ -0,0 +1,342 @@
+  export interface Type {
+    __resolveType?: GraphQLTypeResolver<Student | Professor, Context>;
+    __isTypeOf?: GraphQLIsTypeOfFn<Student | Professor, Context>;
+  }
+}
+
+export namespace QueryResolvers {
+  export const defaultResolvers = {};
+
+  export interface ArgsUsers {
+    first: number | null;
+  }
+
+  export type UsersResolver = (
+    parent: {},
+    args: ArgsUsers,
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => UserType[] | Promise<UserType[]>;
+
+  export interface Type {
+    users: (
+      parent: {},
+      args: ArgsUsers,
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => UserType[] | Promise<UserType[]>;
+  }
+}
+
+export namespace UserResolvers {
+  export const defaultResolvers = {
+    id: (parent: User) => parent.id,
+    name: (parent: User) => parent.name,
+    type: (parent: User) => parent.type
+  };
+
+  export type IdResolver = (
+    parent: User,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type NameResolver = (
+    parent: User,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type TypeResolver = (
+    parent: User,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => UserType | Promise<UserType>;
+
+  export interface Type {
+    id: (
+      parent: User,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    name: (
+      parent: User,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    type: (
+      parent: User,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => UserType | Promise<UserType>;
+  }
+}
+
+export namespace StudentResolvers {
+  export const defaultResolvers = {
+    age: (parent: Student) => parent.age
+  };
+
+  export type AgeResolver = (
+    parent: Student,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => number | Promise<number>;
+
+  export interface Type {
+    age: (
+      parent: Student,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => number | Promise<number>;
+  }
+}
+
+export namespace ProfessorResolvers {
+  export const defaultResolvers = {
+    degree: (parent: Professor) => parent.degree
+  };
+
+  export type DegreeResolver = (
+    parent: Professor,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | null | Promise<string | null>;
+
+  export interface Type {
+    degree: (
+      parent: Professor,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | null | Promise<string | null>;
+  }
+}
+
+export interface Resolvers {
+  Query: QueryResolvers.Type<T>;
+  User: UserResolvers.Type<T>;
+  Student: StudentResolvers.Type<T>;
+  Professor: ProfessorResolvers.Type<T>;
+  UserType?: UserTypeResolvers.Type<T>;
```

Comment:
I currently marked unions and interfaces as optional, but I think this might not be right. Thoughts?